### PR TITLE
Remote access can open/close windoors

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

This PR adds windoor access to remote. 
Remote control can only open/close them, any other actions will result in fail message.

## Why It's Good For The Game

If you can control the electronic door, why can you control electronic windoor?

## Testing

Compiled, tried accessing doors with wrong access, tried multiple modes, tried access tune doors and windoors, tried access depowered windoor

## Changelog
:cl:
tweak: Remote controls(and access tuner) can now access windoors in open/close mode.
/:cl:

### CODE REVIEW

File: code/game/objects/items/control_wand.dm [hunk 1]
```diff
@@ -49,9 +49,13 @@
 	. = ..()
 	. += "<span class='notice'>It's current mode is: [mode]</span>"
 
-/obj/item/door_remote/afterattack(obj/machinery/door/airlock/D, mob/user)
-	if(!istype(D))
-		return
+/obj/item/door_remote/afterattack(obj/target, mob/user) //machinery/door/airlock/D
```
[the remark this answers is not in the document]
Comment (HMBGERDO) [reply code/game/objects/items/control_wand.dm:52]: yeeeep... gonna refactor it to remove duplicated code

**Comment by HMBGERDO** [comment]
yes, duplicated code
i hope thats easier to support and modify than add some internal checks for different objects in one proc

File: code/game/objects/items/control_wand.dm [hunk 2]
```diff
@@ -49,45 +49,85 @@
...
+		if(WAND_BOLT)
+			if(D.locked)
+				D.unlock()
+			else
+				D.lock()
+		if(WAND_EMERGENCY)
+			if(D.emergency)
+				D.emergency = FALSE
+			else
+				D.emergency = TRUE
```
Comment (Sirryan2002) [inline code/game/objects/items/control_wand.dm:95]:
```suggestion
			D.emergency = !D.emergency
```

File: code/game/objects/items/control_wand.dm [hunk 3]
```diff
@@ -49,45 +49,85 @@
...
+		if(WAND_SPEED)
+			D.normalspeed = !D.normalspeed
+			to_chat(user, "<span class='notice'>[D] is now in [D.normalspeed ? "normal" : "fast"] mode.</span>")
+
+/obj/item/door_remote/proc/access_windoor(obj/machinery/door/window/D, mob/user)
+	if(HAS_TRAIT(D, TRAIT_CMAGGED))
+		to_chat(user, "<span class='danger'>The door doesn't respond to [src]!</span>")
+		return
+
+	if(!(D.has_power()))
```
Comment (Sirryan2002) [inline code/game/objects/items/control_wand.dm:106]:
```suggestion
	if(!D.has_power())
```

File: code/game/objects/items/control_wand.dm [hunk 4]
```diff
@@ -49,45 +49,85 @@
...
+		to_chat(user, "<span class='danger'>[D] has no power!</span>")
+		return
+
+	if(!D.requiresID())
+		to_chat(user, "<span class='danger'>[D]'s ID scan is disabled!</span>")
+		return
+
+	if(!D.check_access(src.ID))
```
Comment (Sirryan2002) [inline code/game/objects/items/control_wand.dm:114]:
```suggestion
	if(!D.check_access(ID))
```

File: code/game/objects/items/control_wand.dm [hunk 5]
```diff
@@ -146,7 +186,7 @@
 	var/busy = FALSE
 
 /obj/item/door_remote/omni/access_tuner/afterattack(obj/machinery/door/airlock/D, mob/user)
```
Comment (Sirryan2002) [inline code/game/objects/items/control_wand.dm:188]:
```suggestion
/obj/item/door_remote/omni/access_tuner/afterattack(obj/machinery/door/D, mob/user)
```
need to account for the fact that a non `/airlock` type will be provided as a param here regularly

File: code/game/objects/items/control_wand.dm [hunk 6]
```diff
@@ -189,13 +229,13 @@
 /obj/item/door_remote/janikeyring/afterattack(obj/machinery/door/airlock/D, mob/user, proximity)
```
Comment (Sirryan2002) [inline code/game/objects/items/control_wand.dm:229]: same here

File: code/game/objects/items/control_wand.dm [hunk 7]
```diff
@@ -224,15 +261,30 @@
...
+		D.open()
+	else
+		to_chat(user, "<span class='danger'>The [D] airlock is already open!</span>")
 
+/obj/item/door_remote/janikeyring/access_windoor(obj/machinery/door/window/D, mob/user)
+	if(!(D.has_power()))
+		to_chat(user, "<span class='danger'>[D] has no power!</span>")
+		return
+
+	if(!D.check_access(src.ID))
```
Comment (Sirryan2002) [inline code/game/objects/items/control_wand.dm:279]:
```suggestion
	if(!D.check_access(ID))
```

File: code/game/objects/items/control_wand.dm [hunk 8]
```diff
@@ -224,15 +261,30 @@
...
+		to_chat(user, "<span class='danger'>[src] does not seem to have a key for the [D]'s access panel!</span>")
+		return
+
+	D.add_hiddenprint(user)
+	if(D.density)
+		D.open()
 	else
-		to_chat(user, "<span class='danger'>[src] does not seem to have a key for the [D] airlock's access panel!</span>")
+		to_chat(user, "<span class='danger'>The [D] is already open!</span>")
```
Comment (Sirryan2002) [inline code/game/objects/items/control_wand.dm:287]: I don't know if `The` is needed here

File: code/game/objects/items/control_wand.dm [hunk 9]
```diff
@@ -189,13 +226,13 @@
 /obj/item/door_remote/janikeyring/afterattack(obj/machinery/door/airlock/D, mob/user, proximity)
```
Comment (HMBGERDO) [inline code/game/objects/items/control_wand.dm:226]:
```suggestion
/obj/item/door_remote/janikeyring/afterattack(obj/machinery/door/D, mob/user, proximity)
```

File: code/game/objects/items/control_wand.dm (hunk 8 above)
Comment (HMBGERDO) [reply code/game/objects/items/control_wand.dm:284]: image
is it weird?

Comment (Sirryan2002) [reply code/game/objects/items/control_wand.dm:284]: I remember in the past we removed a lot of instances of `the` b/c string literals would automatically tack on the adjective for some stuff. But frankly it doesn't matter that much.

**Review by Sirryan2002 — APPROVED** [review]

**Pull request merged**